crates/ruma-common/src/events/message/content_serde.rs

```diff
@@ -14,6 +14,14 @@ pub(crate) struct MessageContentSerDeHelper {
     #[serde(rename = "org.matrix.msc1767.text")]
     text_unstable: Option<String>,
 
+    /// HTML short form, stable name.
+    #[serde(rename = "m.html")]
+    html_stable: Option<String>,
+
+    /// HTML short form, unstable name.
+    #[serde(rename = "org.matrix.msc1767.html")]
+    html_unstable: Option<String>,
```

Comment:
What's the reasoning behind having two fields for this?

Also, couldn't we possibly have a `flatten`-ed untagged enum that would catch *either* the stable or unstable one? Because i don't think that both are allowed at the same time?

Reply:
This is because `rename` and `alias` don't work with `flatten` (https://github.com/serde-rs/serde/issues/1504) so we have to deserialize both cases manually.

There's no rule about whether to use the stable or unstable one if both are present afaik. For me, the general rule about deserializing seems to be "ignore the fields you don't know/care about", so if we see the stable we use it, otherwise we try to use the unstable one.

Reply:
Ah, then i think this would be alright, if a bit inconvenient.

Possibly, have a `html(&self) -> Option<&'self str>` method that would pick the "right" html bit, if both are defined.

Reply:
This is just a convenience type for deser, it's not public. There's only one field on the public type.